homeassistant/components/uptimerobot/switch.py

```diff
@@ -0,0 +1,78 @@
+"""UptimeRobot switch platform."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyuptimerobot import UptimeRobotAuthenticationException
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import UptimeRobotDataUpdateCoordinator
+from .const import API_ATTR_OK, DOMAIN, LOGGER
+from .entity import UptimeRobotEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the UptimeRobot switches."""
+    coordinator: UptimeRobotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        UptimeRobotSwitch(
+            coordinator,
+            SwitchEntityDescription(
+                key=str(monitor.id),
+                name=f"{monitor.friendly_name} Active",
+                device_class=SwitchDeviceClass.SWITCH,
+            ),
+            monitor=monitor,
+        )
+        for monitor in coordinator.data
+    )
+
+
+class UptimeRobotSwitch(UptimeRobotEntity, SwitchEntity):
+    """Representation of a UptimeRobot switch."""
+
+    _attr_icon = "mdi:cog"
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the entity is on."""
+        return bool(self.monitor.status == 2)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.monitor.status not in (8, 9)
+
+    async def _async_edit_monitor(self, **kwargs: Any) -> None:
+        """Edit monitor status."""
+        try:
+            response = await self.api.async_edit_monitor(**kwargs)
+        except UptimeRobotAuthenticationException:
+            if self.coordinator.config_entry:
+                self.coordinator.config_entry.async_start_reauth(self.hass)
+            LOGGER.error("API exception: Authentication error with empty config entry")
```

Comment:
empty config entry?

Reply:
Should never happen as config_entry is never `None`.
I logged it anyway, but we can easily drop it.

Reply:
Drop if check, log proper error message.

Reply:
Cannot find a proper way to set config_entry. If I do it on `async_setup_entry` then is too early and later on is still `ConfigEntry | None`.

Reply:
```python
class UptimeRobotDataUpdateCoordinator(...
    ...
    config_entry: ConfigEntry
```

Reply:
Already tried :-(
With:

```python
class UptimeRobotDataUpdateCoordinator(DataUpdateCoordinator):
    """Data update coordinator for UptimeRobot."""

    data: list[UptimeRobotMonitor]
    config_entry: ConfigEntry
```

mypy still complains:

`homeassistant/components/uptimerobot/switch.py:57: error: Item "None" of "Optional[ConfigEntry]" has no attribute "async_start_reauth"  [union-attr]`

Reply:
```python
class UptimeRobotEntity(...
    ....
    coordinator: UptimeRobotDataUpdateCoordinator
```